src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -290,13 +295,28 @@ public void execute() {
 
             // Add implementers to the interface
             if (cu.implementations != null && !cu.implementations.isEmpty()) {
-                for (String i : cu.implementations) {
-                    String fullInterfacePath = determineFullyQualifiedClassName(cu.uses, i);
+                // TempList to avoid conflict
+                List<String> listNewImplement = new ArrayList<>();
+                Iterator<String> iter = cu.implementations.iterator();
+                
+                while(iter.hasNext()) {
+                    String currentImplement = iter.next();
+                    String fullInterfacePath = determineFullyQualifiedClassName(cu.uses, currentImplement);
                     CompilationUnit iface = classes.get(fullInterfacePath);
                     if (iface != null) {
                         iface.implementers.add(cu.name);
+                        // Set FullName for implemented interfaces
+                        if (!currentImplement.equals(fullInterfacePath)){
+                            iter.remove();

```

Comment:
Did you test that ? Removing elements in an iterator in the middle of a loop with next() is usually a bad idea.

Reply:
Yes, it works fine.  
I tested it on my project and I had no problem.

It's why I use an iterator that allow me to remove items safely (unlike a simple foreach). 
But if you don't like it I can find another way (a foreach on a temp list should work as well)

Reply:
I found the trick here : https://stackoverflow.com/questions/1196586/calling-remove-in-foreach-loop-in-java

Reply:
I didn't write the ABLDuck code myself, but this block code is just supposed to replace the interface names with their fully-qualified name ? Could you just add a test case for that ? Code coverage says that this loop is not even tested.

Reply:
I had the test case.